File: frontend/src/types/comment.ts
export interface CommentResponse {
  id: string;
  postId: string;
  parentId?: string | null;
  author: string;
  content: string;
  createdAt: string;
  updatedAt: string;
}

export interface CommentCreateRequest {
  author: string;
  content: string;
  password: string;
  parentId?: string | null;
}

export interface CommentUpdateRequest {
  content: string;
  password: string;
}

export interface CommentListResponse {
  success: boolean;
  comments: CommentResponse[];
  count: number;
}

File: backend/app/schemas/comment.py
from datetime import datetime

from pydantic import Field

from app.schemas.common import CamelModel


class CommentCreate(CamelModel):
    author: str = Field(min_length=1, max_length=50)
    content: str = Field(min_length=1, max_length=5000)
    password: str = Field(min_length=4, max_length=50)


class CommentUpdate(CamelModel):
    content: str = Field(min_length=1, max_length=5000)
    password: str = Field(min_length=4, max_length=50)


class CommentResponse(CamelModel):
    id: str
    post_id: str
    author: str
    content: str
    created_at: datetime
    updated_at: datetime


class CommentListResponse(CamelModel):
    success: bool = True
    comments: list[CommentResponse]
    count: int


class CommentCreateResponse(CamelModel):
    success: bool = True
    message: str
    comment: CommentResponse
    comment_count: int


class CommentUpdateResponse(CamelModel):
    success: bool = True
    message: str
    comment: CommentResponse


class CommentDeleteRequest(CamelModel):
    password: str | None = None

